common/lib/dependabot/metadata_finders/base/changelog_finder.rb

```diff
@@ -319,19 +319,22 @@ def new_version
         end
 
         def previous_ref
-          dependency.previous_requirements.map do |r|
+          previous_refs = dependency.previous_requirements.map do |r|
             r.dig(:source, "ref") || r.dig(:source, :ref)
-          end.compact.first
+          end.compact.uniq
+          return previous_refs.first if previous_refs.count == 1
         end
 
         def new_ref
-          dependency.requirements.map do |r|
+          new_refs = dependency.requirements.map do |r|
             r.dig(:source, "ref") || r.dig(:source, :ref)
-          end.compact.first
+          end.compact.uniq
+          return new_refs.first if new_refs.count == 1
         end
 
         def ref_changed?
-          previous_ref && new_ref && previous_ref != new_ref
+          # We could go from multiple previous refs (nil) to a single new ref
+          previous_ref != new_ref
```

Comment:
I've made this change in all places where we have the same logic but different places have slightly nuanced differences around which one is actually expected to exist so I've moved the presence check out of this method and made it local to wherever it's being used. e.g. `if new_ref && ref_changed? then new_ref else other`